server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -441,12 +441,31 @@ public TSStatus executeBatchStatement(TSExecuteBatchStatementReq req) {
 
       List<TSStatus> result = new ArrayList<>();
       boolean isAllSuccessful = true;
+      InsertRowsPlan insertRowsPlan = new InsertRowsPlan();
+      int index = 0;
       for (String statement : req.getStatements()) {
-        long t2 = System.currentTimeMillis();
-        isAllSuccessful =
-            executeStatementInBatch(statement, result, req.getSessionId()) && isAllSuccessful;
-        Measurement.INSTANCE.addOperationLatency(Operation.EXECUTE_ONE_SQL_IN_BATCH, t2);
+        PhysicalPlan physicalPlan =
+            processor.parseSQLToPhysicalPlan(
+                statement, sessionIdZoneIdMap.get(req.getSessionId()), DEFAULT_FETCH_SIZE);
+        if (physicalPlan.getOperatorType().equals(INSERT)) {
+          insertRowsPlan.addOneInsertRowPlan((InsertRowPlan) physicalPlan, index);
+          index++;
+        } else {
+          long t2 = System.currentTimeMillis();
+          isAllSuccessful =
+              executeStatementInBatch(statement, result, req.getSessionId()) && isAllSuccessful;
```

Comment:
We have parsed the statement to the plan, so we can pass the plan to the `executeStatementInBatch()` method instead of recalled the `parseSQLToPhysicalPlan()` method in executeStatementInBatch().